test/browser/render.js

```diff
@@ -243,209 +284,279 @@ describe('render()', () => {
+				.that.matches(/top\s*:\s*5px\s*/)
+				.and.matches(/position\s*:\s*relative\s*/);
+		});
 
-	it('should only register on* functions as handlers', () => {
-		let click = () => {},
-			onclick = () => {};
+		it('should properly switch from string styles to object styles and back', () => {
+			let root = render((
+				<div style="display: inline;">test</div>
+			), scratch);
 
-		let proto = document.createElement('div').constructor.prototype;
+			expect(root.style.cssText).to.equal('display: inline;');
 
-		sinon.spy(proto, 'addEventListener');
+			root = render((
+				<div style={{ color: 'red' }} />
+			), scratch, root);
 
-		render(<div click={ click } onClick={ onclick } />, scratch);
+			expect(root.style.cssText).to.equal('color: red;');
 
-		expect(scratch.childNodes[0].attributes.length).to.equal(0);
+			root = render((
+				<div style="color: blue" />
+			), scratch, root);
 
-		expect(proto.addEventListener).to.have.been.calledOnce
-			.and.to.have.been.calledWithExactly('click', sinon.match.func, false);
+			expect(root.style.cssText).to.equal('color: blue;');
 
-		proto.addEventListener.restore();
-	});
+			root = render((
+				<div style={{ color: 'yellow' }} />
+			), scratch, root);
+
+			expect(root.style.cssText).to.equal('color: yellow;');
+
+			root = render((
+				<div style="display: block" />
+			), scratch, root);
+
+			expect(root.style.cssText).to.equal('display: block;');
+		});
 
-	it('should add and remove event handlers', () => {
-		let click = sinon.spy(),
-			mousedown = sinon.spy();
+		it('should serialize style objects', () => {
+			let root = render((
+				<div style={{
+					color: 'rgb(255, 255, 255)',
+					background: 'rgb(255, 100, 0)',
+					backgroundPosition: '10px 10px',
+					'background-size': 'cover',
+					padding: 5,
+					top: 100,
+					left: '100%'
+				}}
+				>
+					test
+				</div>
+			), scratch);
 
-		let proto = document.createElement('div').constructor.prototype;
-		sinon.spy(proto, 'addEventListener');
-		sinon.spy(proto, 'removeEventListener');
+			let { style } = scratch.childNodes[0];
+			expect(style).to.have.property('color').that.equals('rgb(255, 255, 255)');
+			expect(style).to.have.property('background').that.contains('rgb(255, 100, 0)');
+			expect(style).to.have.property('backgroundPosition').that.equals('10px 10px');
+			expect(style).to.have.property('backgroundSize', 'cover');
+			expect(style).to.have.property('padding', '5px');
+			expect(style).to.have.property('top', '100px');
+			expect(style).to.have.property('left', '100%');
+
+			root = render((
+				<div style={{ color: 'rgb(0, 255, 255)' }}>test</div>
+			), scratch, root);
+
+			expect(root.style.cssText).to.equal('color: rgb(0, 255, 255);');
+
+			root = render((
+				<div style={{ backgroundColor: 'rgb(0, 255, 255)' }}>test</div>
+			), scratch, root);
+
+			expect(root.style.cssText).to.equal('background-color: rgb(0, 255, 255);');
+		});
+	});
+
+	describe('event handling', () => {
```

Comment:
I grouped all event handling under a common `describe` block

Reply:
I also broke up the event handling tests to better specify the various concerns they were testing

Reply:
Yay, meaningful failure messages!